### PR TITLE
Object fetch changes

### PR DESCRIPTION
**Pull request by piercetrey-figure**
* Object size in bytes in object_permissions table
* Storing object w/ type as the actual object in object-store (and therefore using hash of this proto)

### CODE REVIEW

File: client/src/main/kotlin/io/provenance/objectstore/gateway/client/GatewayClient.kt [hunk 1]
```diff
@@ -104,16 +104,19 @@ class GatewayClient(val config: ClientConfig) : Closeable {
      * @param objectType (optional) the type of data that this represents. This is for reference at the time of retrieval if needed
      * @param jwt a provenance JWT (can be created using this client's `createJwt` methods)
      * @param timeout an optional timeout for the request
+     *
+     * @return a proto containing the hash of the stored object. This hash can be used for future retrieval via [getObject].
```
Comment (hyperschwartz) [inline client/src/main/kotlin/io/provenance/objectstore/gateway/client/GatewayClient.kt:108]: :glorious: documentation

File: server/src/main/kotlin/io/provenance/objectstore/gateway/server/ObjectStoreGatewayServer.kt [hunk 2]
```diff
@@ -39,30 +38,24 @@ class ObjectStoreGatewayServer(
...
         responseObserver: StreamObserver<GatewayOuterClass.PutObjectResponse>
     ) {
-        objectService.putObject(request.objectBytes.toByteArray(), request.type.takeIf { it.isNotBlank() }, publicKey()).let {
-        responseObserver.onNext(
-            GatewayOuterClass.PutObjectResponse.newBuilder()
-                .setHash(it)
-                .build()
-        )
-    }
+        objectService.putObject(request.`object`, publicKey()).let {
```
Comment (hyperschwartz) [inline server/src/main/kotlin/io/provenance/objectstore/gateway/server/ObjectStoreGatewayServer.kt:41]: RIP in reserved words from proto.  IIRC if something gets named `final` things get super weird in Java lol

**Review by hyperschwartz — APPROVED** [review]
Hell YEAH

**Comment by piercetrey-figure** [comment]
for some reason there is a test error (something to do with the object_permissions primary key being null and the in-memory test db)... not sure what is going on, might be fixed by just switching to uuid 🤔

**Comment by hyperschwartz** [comment]
> for some reason there is a test error (something to do with the object_permissions primary key being null and the in-memory test db)... not sure what is going on, might be fixed by just switching to uuid 🤔

Yeah, UUID could be a fixer.  Maybe BigSerial doesn't play nice with SQLite?

File: server/src/main/kotlin/io/provenance/objectstore/gateway/server/ObjectStoreGatewayServer.kt (hunk 2 above)
Comment (piercetrey-figure) [reply server/src/main/kotlin/io/provenance/objectstore/gateway/server/ObjectStoreGatewayServer.kt]: yeah, I thought about changing the field name... but wasn't coming up with anything that sounded good to me

**Pull request merged**